code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -183,12 +211,17 @@
 // Initializes cyborg's components. Technically, adds default set of components to new borgs
 /mob/living/silicon/robot/proc/initialize_components()
 	components["actuator"] = new/datum/robot_component/actuator(src)
+	actuatorComponent = components["actuator"]
 	components["radio"] = new/datum/robot_component/radio(src)
 	components["power cell"] = new/datum/robot_component/cell(src)
 	components["diagnosis unit"] = new/datum/robot_component/diagnosis_unit(src)
 	components["camera"] = new/datum/robot_component/camera(src)
 	components["comms"] = new/datum/robot_component/binary_communication(src)
 	components["armour"] = new/datum/robot_component/armour(src)
+	components["jetpack"] = new/datum/robot_component/jetpack(src)
+	jetpackComponent = components["jetpack"]
+	var/datum/robot_component/JC = components["jetpack"]
+	JC.installed = 0//We start the jetpack as not installed, because its nondefault

```

Comment:
This is the way all components work. Components are never removed from the components list, they're simply toggled to uninstalled when removed, and spawn the external item. Changing it would be lots of work for little benefit, to remove various reference errors that would result, and making the jetpack be a wierd exception runs counter to why i did this in the first place